src/helpers/animation/index.js

```diff
@@ -0,0 +1,44 @@
+import { css } from 'styled-components'
+
+import { circleGrow } from './circleGrow'
+import { flash } from './flash'
+import { focus } from './focus'
+import { shine } from './shine'
+import {
+  slideBottom,
+  slideLeft,
+  slideOpenDiag,
+  slideOpenHorizontal,
+  slideOpenVertical,
+  slideRight,
+  slideTop,
+} from './slide'
+
+const baseCss = css`
+  align-items: center;
+  display: inline-flex;
+  justify-content: center;
+  overflow: hidden;
+  position: relative;
+  &:after {
+    content: '';
+    position: absolute;
+  }
+`
+
+const getAnimation = type => css`
+  ${baseCss};
+  ${type === 'circleGrow' && circleGrow};
```

Comment:
I think you can simply this with `eval` and do : 

```
 ${eval(type)}
```

Reply:
Using `eval` might simplify a lot the code, but it can brings potential security issues caused by code injection. If we follow the trace, `getAnimation(type)` receive a prop, which means we don't have a total control over the value.

However, I agree with you that using `eval` here would be very convenient as it would give a generic solution to the animations, instead of listing them one by one.